pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -165,12 +165,13 @@ func (r *components) renderComponent(ctx context.Context, acc v1alpha2.Applicati
 }
 
 func (r *components) renderTrait(ctx context.Context, ct v1alpha2.ComponentTrait, namespace, componentName string, ref *metav1.OwnerReference, dag *dependency.DAG) (*unstructured.Unstructured, *v1alpha2.TraitDefinition, error) {
+	traitName := util.GenTraitName(componentName, ct.DeepCopy(), nil)
```

Comment:
I think we must check if there's already an exist trait, before generate a new traitname

Reply:
done